### DISCUSSION TITLE
Multiple Lotteries Lesson 7

### DISCUSSION THREAD

**Discussion by EcologicWind** [smartcontractkit/full-blockchain-solidity-course-py#911] [Ideas]
Hello everyone, I was thinking about what is the best way to approach making multiple lottery pools? 

1.  Easiest way would be to deploy different contracts and interact with them with the UI. 
2.  Using the same lottery contract expand everything to array form and for entering the lottery and starting the lottery passing the index of the lotto pool. 
3.  I'm not sure how this works but making a lotto struct. 

I look forward to hearing your ideas.

**Comment by cromewar** [comment]
You can create a contract Factory, and just change the specifics you need for each lottery pool,
Here you have a example of the factory pattern: 

https://betterprogramming.pub/learn-solidity-the-factory-pattern-75d11c3e7d29

**Reply by EcologicWind** [reply]
This looks very interesting. Thank you @cromewar I will try it and post the results if I succeed.
